software/base/src/main/java/org/apache/brooklyn/entity/software/base/lifecycle/MachineLifecycleEffectorTasks.java

```diff
@@ -1083,4 +1146,39 @@ protected void clearEntityLocationAttributes(Location machine) {
         entity().sensors().set(Attributes.SUBNET_ADDRESS, null);
     }
 
+    public static ReleaseableLatch waitForLatch(EntityInternal entity, ConfigKey<Boolean> configKey) {
+        Maybe<?> rawValue = entity.config().getRaw(configKey);
+        if (rawValue.isAbsent()) {
+            return ReleaseableLatch.NOP;
+        } else {
+            ValueResolverIterator<Boolean> iter = resolveLatchIterator(entity, rawValue.get(), configKey);
+            Maybe<ReleaseableLatch> releasableLatchMaybe = iter.next(ReleaseableLatch.class);
```

Comment:
`next(Class<?> type)` will return the first element instance of `type` - it's doing the looping under the hood.